### PR TITLE
Add version returning MegaError pointer instead of MegaError

### PR DESCRIPTION
**Pull request by alber2510**

### CODE REVIEW

File: src/megaapi_impl.cpp [hunk 1]
```diff
@@ -16399,17 +16399,23 @@ void MegaApiImpl::processTransferRemoved(Transfer *tr, MegaTransferPrivate *tran
 }
 
 MegaError MegaApiImpl::checkAccess(MegaNode* megaNode, int level)
+{
+    std::unique_ptr<MegaError>megaError = std::unique_ptr<MegaError>(checkAccessErrorExtended(megaNode, level));
```
Comment (sergiohs84) [inline src/megaapi_impl.cpp:16403]:
```suggestion
    std::unique_ptr<MegaError> megaError(checkAccessErrorExtended(megaNode, level));
```

File: src/megaapi_impl.cpp [hunk 2]
```diff
@@ -16430,22 +16436,28 @@ MegaError MegaApiImpl::checkAccess(MegaNode* megaNode, int level)
...
     }
 
-    return client->checkaccess(node, a) ? API_OK : API_EACCESS;
+    return client->checkaccess(node, a) ? new MegaErrorPrivate(API_OK) : new MegaErrorPrivate(API_EACCESS);
 }
 
 MegaError MegaApiImpl::checkMove(MegaNode* megaNode, MegaNode* targetNode)
 {
-    if(!megaNode || !targetNode) return API_EARGS;
+    std::unique_ptr<MegaError>megaError = std::unique_ptr<MegaError>(checkMoveErrorExtended(megaNode, targetNode));
```
Comment (sergiohs84) [inline src/megaapi_impl.cpp:16444]:
```suggestion
    std::unique_ptr<MegaError> megaError(checkMoveErrorExtended(megaNode, targetNode));
```

File: include/megaapi.h [hunk 3]
```diff
@@ -13628,6 +13628,28 @@ class MegaApi
...
+         * - MegaShare::ACCESS_READ
+         *
+         * @return Pointer to MegaError with the result.
+         * Valid values for the error code are:
+         * - MegaError::API_OK - The node has the required access level
+         * - MegaError::API_EACCESS - The node doesn't have the required access level
+         * - MegaError::API_ENOENT - The node doesn't exist in the account
+         * - MegaError::API_EARGS - Invalid parameters
+         */
+        MegaError* checkAccessErrorExtended(MegaNode* node, int level);
```
Comment (sergiohs84) [inline include/megaapi.h:13651]: Should we tag the old interfaces `checkAccess()` and `checkMove()` as deprecated??

Comment (alber2510) [reply include/megaapi.h:13651]: Yes, probably would be good, I'll add it

**Pull request merged**